src/form/FileUpload/FileList/FileList.tsx

```diff
@@ -1,70 +1,110 @@
+	FileListItemImagePreviewStyle,
+	FileListItemPreviewWrapperStyle,
 	FileListStyle,
 	FileListItemStyle,
 	FileNameStyle,
 	FileNameErrorStyle,
 	FileProgressBarStyle,
+	FileListTitleStyle,
 } from './FileListStyles';
-import { CustomFile, Files } from '../hooks';
-import { FileRejection } from 'react-dropzone';
+import type { CustomFile, CustomFileOrRejection, Files } from '../hooks';
 
-export type Props = {
+export type FileListProps = {
 	files: Files;
 	removeLabel: string;
 	cancelLabel: string;
-	onFileRemove: (file: CustomFile & FileRejection) => void;
-	onCancel: (file: CustomFile & FileRejection) => void;
+	onFileRemove: (file: CustomFileOrRejection) => void;
+	onCancel: (file: CustomFileOrRejection) => void;
 	fileUploadErrorLabel: string;
-	fileUploadInProgressLabel: string;
+	fileUploadInProgressLabel?: string;
 	isUploading?: boolean;
 	progress?: number;
+	title?: string;
 };
 
 type FileListItemProps = {
-	file: CustomFile & FileRejection;
+	file: CustomFileOrRejection;
 	removeLabel: string;
 	cancelLabel: string;
 	onFileRemove: (file: CustomFile) => void;
 	onCancel: (file: CustomFile) => void;
 	fileUploadErrorLabel: string;
-	fileUploadInProgressLabel: string;
+	fileUploadInProgressLabel?: string;
 };
 
-function isFileUploading(file: CustomFile & FileRejection) {
-	return file && file.progress && file.progress > 0 && file.progress < 100;
-}
+const isFileUploading = (file: CustomFileOrRejection) => (file && file.progress && file.progress < 100 ? true : false);
+
+const isFileUploadingIndeterminate = (file: CustomFileOrRejection) => (file && file.progress === 0 ? true : false);
 
-const FileList: React.FC<Props> = ({
+const FileList: React.FC<FileListProps> = ({
 	files,
 	cancelLabel,
 	removeLabel,
 	onFileRemove,
 	onCancel,
 	fileUploadErrorLabel,
 	fileUploadInProgressLabel,
+	title,
 	...otherProps
-}: Props) => (
-	<FileListStyle data-testid="file-list" {...otherProps}>
-		{files.map((file, index) => (
-			<FileListItem
-				key={file.tmpId || index}
-				onCancel={() => onCancel(file)}
-				onFileRemove={() => onFileRemove(file)}
-				cancelLabel={cancelLabel}
-				removeLabel={removeLabel}
-				file={file}
-				fileUploadErrorLabel={fileUploadErrorLabel}
-				fileUploadInProgressLabel={fileUploadInProgressLabel}
-			/>
-		))}
-	</FileListStyle>
-);
+}) => {
+	if (files?.length === 0) {
+		return null;
+	}
+
+	return (
+		<FileListStyle data-testid="file-list" {...otherProps}>
+			{title && <FileListTitleStyle>{title}</FileListTitleStyle>}
+			{files.map((file, index) => (
+				<FileListItem
+					key={file.tmpId || index}
+					onCancel={() => onCancel(file)}
+					onFileRemove={() => onFileRemove(file)}
+					cancelLabel={cancelLabel}
+					removeLabel={removeLabel}
+					file={file}
+					fileUploadErrorLabel={fileUploadErrorLabel}
+					fileUploadInProgressLabel={fileUploadInProgressLabel}
+				/>
+			))}
+		</FileListStyle>
+	);
+};
+
+const useFileReader = (setPreview: (val: string) => void) => {
```

Comment:
Would it be a bit cleaner just have `file` as a dependency/param? And the hook just exposes the result? So you have all the  filereader logic in the hook instead in the implementation side? Something like
`const {result: preview} = useFileReader(file)`.
Would also move it to it's own file, so you can test it :)